src/Standards/Generic/Sniffs/ControlStructures/ControlStructuresBracketsNewLineSniff.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * Checks if Control Structures Brackets are on a line by their own
+ *
+ * @author    Greg Sherwood <gsherwood@squiz.net>
+ * @copyright 2006-2018 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\ControlStructures;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+class ControlStructuresBracketsNewLineSniff implements Sniff
+{
+
+    /**
+     * The number of spaces code should be indented.
+     *
+     * @var integer
+     */
+    public $indent = 4;
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            T_IF,
+            T_ELSEIF,
+            T_ELSE,
+            T_FOREACH,
+            T_FOR,
+            T_SWITCH,
+            T_DO,
+            T_WHILE,
+            T_TRY,
+            T_CATCH,
+            T_FINALLY,
+        ];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param integer                     $stackPtr  The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens      = $phpcsFile->getTokens();
+        $errorData   = [strtolower($tokens[$stackPtr]['content'])];
+        $openBrace   = $tokens[$stackPtr]['scope_opener'];
+        $lastContent = $phpcsFile->findPrevious(T_WHITESPACE, ($openBrace - 1), $stackPtr, true);
+        $controlStructureLine = $tokens[$lastContent]['line'];
+        $braceLine            = $tokens[$openBrace]['line'];
+
+        if ($braceLine === $controlStructureLine) {
+            $phpcsFile->recordMetric($stackPtr, 'Control Structure opening brace placement', 'same line');
+            $error = 'Opening brace of a %s must be on the line after the definition';
+            $fix   = $phpcsFile->addFixableError($error, $openBrace, 'OpenBraceNewLine', $errorData);
+
+            if ($fix === true) {
+                $phpcsFile->fixer->beginChangeset();
+
+                if ($tokens[($openBrace - 1)]['code'] === T_WHITESPACE) {
+                    $phpcsFile->fixer->replaceToken(($openBrace - 1), '');
+                }
+
+                $phpcsFile->fixer->addNewlineBefore($openBrace);
+                $phpcsFile->fixer->endChangeset();
+            }
+
+            return;
+        } else {
+            $phpcsFile->recordMetric($stackPtr, 'Control Structure opening brace placement', 'new line');
+
+            if ($braceLine > ($controlStructureLine + 1)) {
+                $error = 'Opening brace of a %s must be on the line following the %s declaration.; Found %s line(s).';
```

Comment:
Sniff errors don't end with full stops
```suggestion
                $error = 'Opening brace of a %s must be on the line following the %s declaration; found %s line(s)';
```